alacritty_terminal/src/term/mod.rs

```diff
@@ -1875,8 +1890,22 @@ impl<T: EventListener> Handler for Term<T> {
                     let template = Cell { bg: template.bg, ..Cell::default() };
                     self.grid.clear_viewport(template);
                 }
+
+                self.selection =
+                    self.selection.take().filter(|s| !s.intersects_range(0..=last_line));
+            },
+            ansi::ClearMode::Saved => {
+                let last_scrollback_line = self.grid.len() - 1;
+
+                self.grid.clear_history();
+
+                if last_scrollback_line > last_line {
+                    self.selection = self
+                        .selection
+                        .take()
+                        .filter(|s| !s.intersects_range(last_line + 1..=last_scrollback_line));
+                }
```

Comment:
Why the conditional here instead of the match arm? We can completely skip this entire thing if there is no history, right?

Reply:
>Why the conditional here instead of the match arm? We can completely skip this entire thing if there is no history, right?

RIght, but then you have non exhaustive match, which looks kind of redundant and could be confusing and error prone sometimes(since you'll basically create a rule `_ => ()`). If anything we can guard in arm at the start.